internal/cmd/byok/command_register.go

```diff
@@ -0,0 +1,222 @@
+package byok
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"regexp"
+
+	"github.com/spf13/cobra"
+
+	byokv1 "github.com/confluentinc/ccloud-sdk-go-v2/byok/v1"
+	pcmd "github.com/confluentinc/cli/internal/pkg/cmd"
+	"github.com/confluentinc/cli/internal/pkg/errors"
+	"github.com/confluentinc/cli/internal/pkg/examples"
+	"github.com/confluentinc/cli/internal/pkg/output"
+	"github.com/confluentinc/cli/internal/pkg/resource"
+	"github.com/confluentinc/cli/internal/pkg/utils"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+)
+
+var encryptionKeyPolicyAWS = template.Must(template.New("encryptionKeyPolicyAWS").Parse(`{
+	"Sid" : "Allow Confluent accounts to use the key",
+	"Effect" : "Allow",
+	"Principal" : {
+		"AWS" : [{{range $i, $e := .}}{{if $i}},{{end}}
+			"{{$e}}"{{end}}
+		]
+	},
+	"Action" : [ "kms:Encrypt", "kms:Decrypt", "kms:ReEncrypt*", "kms:GenerateDataKey*", "kms:DescribeKey" ],
+	"Resource" : "*"
+}, {
+	"Sid" : "Allow Confluent accounts to attach persistent resources",
+	"Effect" : "Allow",
+	"Principal" : {
+		"AWS" : [{{range $i, $e := .}}{{if $i}},{{end}}
+			"{{$e}}"{{end}}
+		]
+	},
+	"Action" : [ "kms:CreateGrant", "kms:ListGrants", "kms:RevokeGrant" ],
+	"Resource" : "*"
+}`))
+
+const keyVaultCryptoServiceEncryptionUser = "e147488a-f6f5-4113-8e2d-b22465e65bf6"
+const keyVaultReader = "21090545-7ca7-4776-b22c-e363652d74d2"
```

Comment:
What do these signify? Can they change?

Reply:
These are built-in Azure roles to control access to Azure resources. They don't change:

https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#key-vault-crypto-service-encryption-user

https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#key-vault-reader

Reply:
Yet another interesting design decision made by Microsoft...